Whitelisting.py

```diff
@@ -375,6 +376,14 @@ def check(self, pkg):
                 printError(pkg, self.m_error_map['unauthorized'], f)
                 continue
 
+            # avoid testing the same paths multiple times thereby avoiding
+            # duplicate error messages or unnecessary re-checks of the same
+            # files.
+            # this is necessary since whitelisting entries can consist of
+            # groups of files that are all checked in one go below.
+            if f in already_tested:
```

Comment:
Shouldn't this kind of check be done after resolving symlinks? Otherwise we still get duplicate results if multiple links point to the same file...

Reply:
True, I will try to adjust, if it's easily possible at all.

Reply:
I've been thinking more about this, I think it's not a big deal not to add the symlink targets here. The whitelisting should be rather symlink agnostic, it doesn't care whether a path is a symlink or a regular file, the content must match. So packagers can change file types over time but not the content.

On a logical level, when there is a whitelisting entry like in `sarg`, where now three symlinks to the same file are installed in `/etc/cron.*` then I think it's correct to print the error for each of these symlink paths even though they ultimately point to the same file.

The problem I originally fixed with this commit was that in the case just described the three error messages for the three symlinks have been printed three times, which is clearly unintended.

Reply:
OK, now I understand this a bit more. I agree on what we want to achieve here. (I also didn't read the code carefully enough and thought this was all about symlinks.)

The nested loops still seem a bit convoluted to me. Why not do something like:

1. collect list of cronjobs
2. compare list against ghost files -> print errors
3. compare list against whitelisted files for the package:
   - if whitelist exists for unpackaged file: print non-fatal lint to please make issue in [rpmlint-security-whitelistings](https://github.com/openSUSE/rpmlint-security-whitelistings) to remove whitelisting
   - if packaged file is not whitelisted or checksum mismatch: print fatal lint

The json with cronjob whitelists is already organized by package name - rather than making a new datastructure to allow lookup by path and then filtering the results by package we can simplify things overall by doing it this way.

Reply:
We've decided off-band to postpone this kind of change to another PR.